src/main/java/com/google/swarm/tokenization/DLPTextToBigQueryStreamingV2.java

```diff
@@ -67,64 +62,100 @@ public static void main(String[] args) {
   }
 
   public static PipelineResult run(DLPTextToBigQueryStreamingV2PipelineOptions options) {
-
     Pipeline p = Pipeline.create(options);
+
     switch (options.getDLPMethod()) {
       case INSPECT:
       case DEID:
-        PCollection<KV<String, ReadableFile>> inputFile =
-            p.apply(
-                "CSVReaderTransform",
-                CSVReaderTransform.newBuilder()
-                    .setDelimeter(options.getDelimeter())
-                    .setFilePattern(options.getCSVFilePattern())
-                    .setKeyRange(options.getKeyRange())
-                    .setInterval(DEFAULT_POLL_INTERVAL)
-                    .build());
 
-        final PCollectionView<List<String>> header =
-            inputFile
-                .apply(
-                    "GlobalWindow",
-                    Window.<KV<String, ReadableFile>>into(new GlobalWindows())
-                        .triggering(
-                            Repeatedly.forever(AfterProcessingTime.pastFirstElementInPane()))
-                        .discardingFiredPanes())
-                .apply("ReadHeader", ParDo.of(new CSVFileHeaderDoFn()))
-                .apply("ViewAsList", View.asList());
+        PCollection<KV<String, Table.Row>> inputRows;
+        PCollection<KV<String, ReadableFile>> inputFiles;
+        final PCollectionView<List<String>> header;
 
-        PCollectionTuple transformedContents =
-            inputFile
-                .apply(
-                    "ReadFile",
-                    ParDo.of(
-                        new FileReaderSplitDoFn(options.getKeyRange(), options.getDelimeter())))
-                .apply(
-                    "ConvertDLPRow", ParDo.of(new MapStringToDlpRow(options.getColumnDelimeter())))
-                .apply(
-                    "Fixed Window",
-                    Window.<KV<String, Table.Row>>into(FixedWindows.of(WINDOW_INTERVAL)))
+        // Figure out which file type to use for the pipeline
+        Util.FileType fileType = options.getFileType();
+        // Override file type if the filePattern option contains a file extension
+        if (options.getFilePattern().toLowerCase().endsWith(".csv")) {
+          fileType = Util.FileType.CSV;
+        }
+        if (options.getFilePattern().toLowerCase().endsWith(".avro")) {
+          fileType = Util.FileType.AVRO;
+        }
```

Comment:
Yes that makes sense. I've just pushed the change.

Reply:
One thing though: I had to change the type from `Enum` to `String`, otherwise I got a warning that said I was using a `raw type of Enum`. Let me know if you think that should be handled differently.